projects/rulenode-core-config/src/lib/components/common/select-attributes.component.ts

```diff
@@ -35,23 +46,45 @@ export class SelectAttributesComponent implements  OnInit, ControlValueAccessor,
 
   ngOnInit(): void {
     this.attributeControlGroup = this.fb.group({
-      clientAttributeNames: [null, []],
-      sharedAttributeNames: [null, []],
-      serverAttributeNames: [ null, []],
-      latestTsKeyNames: [null, []],
+      keyFields: this.fb.group({
+        clientAttributeNames: [null, []],
+        sharedAttributeNames: [null, []],
+        serverAttributeNames: [null, []],
+        latestTsKeyNames: [null, []]
+      }, {validators: this.atLeastOneValidator()}),
```

Comment:
What is the point of changing the structure of the form? You need to write the validator correctly.